lib/schema/validate.js

```diff
@@ -56,7 +56,7 @@ const ORIGINAL_SCHEMA = 'original_schema';
  *
  * @returns Intermediate error and warning caches along with stats. Can be combined later using `combineErrors`
  */
-const validate = ({
+const validateJSONPayload = ({
   jsonSchema = {},
   resourceName = 'Property',
```

Comment:
`resourceName` should not default to string here. If no value is provided and `resourceName` is required, then an error should be thrown.